test/index.js

```diff
@@ -1,472 +1,112 @@
-  public: fs.readFileSync(__dirname + '/ec.p192.pub')
-}
+var fixtures = require('./fixtures')
+var myCrypto = require('../browser')
+var nodeCrypto = require('crypto')
+var parseKeys = require('parse-asn1')
+var test = require('tape')
 
-var ec256 = {
-  private: fs.readFileSync(__dirname + '/ec.p256.priv'),
-  public: fs.readFileSync(__dirname + '/ec.p256.pub')
-}
-var ecpass = {
-  private: {
-    key: fs.readFileSync(__dirname + '/ec.pass.priv'),
-    passphrase: 'bard'
-  },
-  public: fs.readFileSync(__dirname + '/ec.pub')
-}
-var dsa = {
-  private: fs.readFileSync(__dirname + '/dsa.1024.priv'),
-  public: fs.readFileSync(__dirname + '/dsa.1024.pub')
-}
-var dsa2 = {
-  private: fs.readFileSync(__dirname + '/dsa.2048.priv'),
-  public: fs.readFileSync(__dirname + '/dsa.2048.pub')
-}
-var dsapass = {
-  private: {
-    key: fs.readFileSync(__dirname + '/pass.dsa.1024.priv'),
-    passphrase: 'password'
-  },
-  public: fs.readFileSync(__dirname + '/pass.dsa.1024.pub')
-}
-var dsapass2 = {
-  private: {
-    key: fs.readFileSync(__dirname + '/pass2.dsa.1024.priv'),
-    passphrase: 'password'
-  },
-  public: fs.readFileSync(__dirname + '/pass2.dsa.1024.pub')
-}
-var rsapass = {
-  private: {
-    key: fs.readFileSync(__dirname + '/pass.rsa.1024.priv'),
-    passphrase: 'password'
-  },
-  public: fs.readFileSync(__dirname + '/pass.rsa.1024.pub')
-}
-var rsapass2 = {
-  private: {
-    key: fs.readFileSync(__dirname + '/pass.rsa.2028.priv'),
-    passphrase: 'password'
-  },
-  public: fs.readFileSync(__dirname + '/pass.rsa.2028.pub')
-}
-var vector = fs.readFileSync(__dirname + '/vector.priv')
-var vector2 = fs.readFileSync(__dirname + '/vector2.priv')
 function isNode10 () {
   return process.version && process.version.split('.').length === 3 && parseInt(process.version.split('.')[1], 10) <= 10
 }
-var nodeCrypto = require('crypto')
-var myCrypto = require('../browser')
-function testIt (keys, message, scheme) {
-  var pub = keys.public
-  var priv = keys.private
+
+fixtures.valid.rsa.forEach(function(f) {
+  var message = new Buffer(f.message)
+  var pub = new Buffer(f.public, 'base64')
+  var priv
+
+  if (f.passphrase) {
+    priv = {
+      key: new Buffer(f.private, 'base64'),
+      passphrase: f.passphrase
+    }
+  } else {
+    priv = new Buffer(f.private, 'base64')
+  }
+
   test(message.toString(), function (t) {
-    t.plan(4)
-    var mySign = myCrypto.createSign(scheme)
-    var nodeSign = nodeCrypto.createSign(scheme)
+    t.plan(5)
+
+    var mySign = myCrypto.createSign(f.scheme)
+    var nodeSign = nodeCrypto.createSign(f.scheme)
     var mySig = mySign.update(message).sign(priv)
     var nodeSig = nodeSign.update(message).sign(priv)
+
     t.equals(mySig.length, nodeSig.length, 'correct length')
     t.equals(mySig.toString('hex'), nodeSig.toString('hex'), 'equal sigs')
-    var myVer = myCrypto.createVerify(scheme)
-    var nodeVer = nodeCrypto.createVerify(scheme)
+    t.equals(mySig.toString('hex'), f.signature)
```

Comment:
RSA is always deterministic, the DSA ones were originally non deterministic
but are now.

On Tue, Mar 10, 2015, 7:32 PM Daniel Cousens <notifications@github.com>
wrote:

> In test/index.js
> <https://github.com/crypto-browserify/browserify-sign/pull/11#discussion_r26177164>
> :
>
> >      t.equals(mySig.length, nodeSig.length, 'correct length')
> >      t.equals(mySig.toString('hex'), nodeSig.toString('hex'), 'equal sigs')
> > -    var myVer = myCrypto.createVerify(scheme)
> > -    var nodeVer = nodeCrypto.createVerify(scheme)
> > +    t.equals(mySig.toString('hex'), f.signature)
>
> I guess the RSA signatures are deterministic?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/crypto-browserify/browserify-sign/pull/11/files#r26177164>
> .
>